src/validation.cairo

```diff
@@ -73,8 +73,18 @@ fn validate_timestamp(self: @ChainState, block: @Block) -> Result<(), ByteArray>
 }
 
 fn next_prev_timestamps(self: @ChainState, block: @Block) -> Span<u32> {
-    // TODO: implement
-    *self.prev_timestamps
+    let mut timestamps = ArrayTrait::<u32>::new();
```

Comment:
You should rather use Array type, which has `append_span`, `pop_front` and `append` functions.